src/tests/unit/tests/common/state-dispatcher.test.ts

```diff
@@ -80,16 +84,69 @@ describe('StateDispatcherTest', () => {
                 privateDispatcher = action;
             });
 
-        const defaultBoardcastMessage = (message: Object) => {};
-        const broadcastMock = Mock.ofInstance<(message: Object) => void>(defaultBoardcastMessage);
+        const broadcastMock = Mock.ofType<(message: Object) => Promise<void>>();
+        broadcastMock.setup(m => m(It.isAny())).returns(() => Promise.resolve());
 
         const stateDispatcher = new StateDispatcher(broadcastMock.object, storeHubMock.object);
         stateDispatcher.initialize();
+
+        broadcastMock.reset();
+        broadcastMock
+            .setup(m => m(expectedMessage))
+            .returns(() => Promise.resolve())
+            .verifiable(Times.once());
+
+        privateDispatcher.call(stateDispatcher);
+
+        broadcastMock.verifyAll();
+    });
+
+    test('propagate exceptions in broadcasting changes to logger.error', async () => {
+        const newstoreData: StoreStubData = { value: 'testValue' };
+        const expectedMessage: StoreUpdateMessage<StoreStubData> = {
+            isStoreUpdateMessage: true,
+            type: GenericStoreMessageTypes.storeStateChanged,
+            storeId: 'testStoreId',
+            storeType: StoreType.TabContextStore,
+            payload: newstoreData,
+        };
+
+        let privateDispatcher: Function;
+        const storeMock: IMock<BaseStore<StoreStubData>> = Mock.ofType<BaseStore<StoreStubData>>();
+        const storeHubMock = Mock.ofType<StoreHubStub>(null, MockBehavior.Strict);
+
+        storeHubMock.setup(x => x.getAllStores()).returns(() => [storeMock.object]);
```

Comment:
don't love `x` to be the name of the mock instance... could be `hub` if you'd like some short name